### PR TITLE
Tdl 13149 sync payment intent

### PR DESCRIPTION
**Pull request by prijendev**
# Description of change
- Added a new stream payment_intents has been added to the tap.
- Schema for payment_intents has been added.
- The added schema for charge stream in the shared directory as it is being used by payment_intent schema.
- Updated integration test cases for payment_intents stream.

# Manual QA steps
 - Verify that schema and metadata are available in the catalog for payment_intents by running discover mode.
 - Verify that data for payment_intents was collected by running sync mode.
 - Verify that payment_intents respect a bookmark.
 
# Risks
 - 
 
# Rollback steps
 - revert this branch

### CODE REVIEW

File: tests/test_bookmarks.py [hunk 1]
```diff
@@ -271,13 +272,21 @@ def test_run(self):
...
-                    expected_updated_key = 'metadata'
-                    expected_updated_value_substring = 'bob'
-                    updated_pk_value = updated_record.get('id')
-                    sync_records_metadata = [sync_record.get('metadata')
-                                             for sync_record in second_sync_data
-                                             if sync_record.get('id') == updated_pk_value]
-                    self.assertTrue(len(sync_records_metadata) == 1)
-                    self.assertIn(expected_updated_value_substring,
-                                  sync_records_metadata[0].get('test_value'))
+                if stream == "payment_intents":
```
Comment (dbshah1212) [inline tests/test_bookmarks.py:275]: Add code comment, why we need to add this if-else.

Comment (prijendev) [reply tests/test_bookmarks.py:275]: Added comments

File: tap_stripe/schemas/payment_intents.json [hunk 2]
```diff
@@ -0,0 +1 @@
+{"$ref": "shared/payment_intent.json#/"}
```
[the remark this answers is not in the document]
Comment (prijendev) [reply tap_stripe/schemas/payment_intents.json]: The removed schema was inside the shared folder. The code generates catalog by schema directly available in the `schemas` folder.

File: tap_stripe/schemas/charges.json [hunk 3]
```diff
@@ -1,1776 +1 @@
...
-    "updated": {
-      "type": [
-        "null",
-        "string"
-      ],
-      "format": "date-time"
-    }
-  }
-}
+{"$ref": "shared/charge.json#/"}
```
Comment (prijendev) [inline tap_stripe/schemas/charges.json:1]: Moved schema for charges into `shared` folder as it is being used in the schema of newly added stream `payment_intents`.

**Review by savan-chovatiya — APPROVED** [review]

File: tap_stripe/schemas/payment_intents.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (dbshah1212) [reply tap_stripe/schemas/payment_intents.json]: Can't we use the payment_intent.json which is inside the shared folder?

[the remark this answers is not in the document]
Comment (prijendev) [reply tap_stripe/schemas/payment_intents.json]: The existing code directly looks into the `schemas` folder for each stream to load the schema. [Reference](https://github.com/singer-io/tap-stripe/blob/master/tap_stripe/__init__.py#L258).  Then, resolve  [`refs`](https://github.com/singer-io/tap-stripe/blob/master/tap_stripe/__init__.py#L291)  for the schema of each stream. For the example the `plan` field is available in the `invoice_items`, `invoice_line_items` schema.  So, the schema for `plan` is available in the shared folder and it is being used in `invoice_items`, `invoice_line_items`, `plans` schema available in the `schemas` folder. [Reference](https://github.com/singer-io/tap-stripe/blob/master/tap_stripe/schemas/plans.json#L1). That's why we kept the schema for `payment_intents` in the `schemas` folder.

**Review by dbshah1212 — APPROVED** [review]

**Review by kspeer825 — APPROVED** [review]

**Review by RushT007 — APPROVED** [review]
Looks good 👍

**Review by KrisPersonal — APPROVED** [review]

**Pull request merged**